Source/Orts.Simulation/Simulation/RollingStocks/MSTSDieselLocomotive.cs

```diff
@@ -662,6 +650,24 @@ protected override void UpdateTractiveForce(float elapsedClockSeconds, float t,
                     w = 0;
                 AverageForceN = w * AverageForceN + (1 - w) * TractiveForceN;
             }
+
+            // Calculate fuel consumption will occur unless diesel engine is stopped
+            if (LocomotivePowerSupply.MainPowerSupplyOn)
```

Comment:
Main power supply on means that power is available for traction, not that the engine is running.
You should use diesel engine state.

Reply:
Ok, I have removed this condition as the diesel engine state should manage diesel fuel flow.

Reply:
The functionnality is correct, but the indentation needs to be fixed.

Reply:
Indentation fixed

Reply:
Thank you. It's approved.